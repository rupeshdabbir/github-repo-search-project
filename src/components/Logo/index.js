// @flow

import React from 'react';
import { Link } from 'react-router-dom';
import { StyleSheet, css } from 'aphrodite/no-important';
import 'suitcss-utils-flex/lib/flex.css';
import GitHubIcon from './mark-github.svg';

export default function Logo() {
  return (
    <div>
      <h1 className={css(styles.Logo_title)}>
        <Link to="/" className={`${css(styles.Logo_link)} u-flex`}>
          <GitHubIcon className={css(styles.Logo_icon)} width={24} height={24} />
          GitHub Repo Search
        </Link>
      </h1>
      <p className={css(styles.Logo_subTitle)}>
        Browse repos via <a href="https://developer.github.com/v3/">the GitHub API</a>
      </p>
    </div>
  );
}

const styles = StyleSheet.create({
  Logo_icon: {
    marginRight: 8,
  },

  Logo_title: {
    fontSize: 25,
  },

  Logo_link: {
    textDecoration: 'none',
    color: 'inherit',
  },

  Logo_subTitle: {
    marginTop: 5,
    fontSize: 14,
  },
});
