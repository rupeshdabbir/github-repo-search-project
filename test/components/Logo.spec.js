import React from 'react';
import renderer from 'react-test-renderer';
import {
  StyleSheetTestUtils,
} from 'aphrodite';
import {
  StaticRouter as Router,
} from 'react-router-dom';

import Logo from 'components/Logo';

describe('Component: Logo', () => {

  beforeAll(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterAll(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('should render', () => {
    const component = renderer.create(
      <Router context={{}}>
        <Logo />
      </Router>
    );
    expect(component).toMatchSnapshot();
  });

});
