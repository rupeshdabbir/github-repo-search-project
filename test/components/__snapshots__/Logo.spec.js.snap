// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Logo should render 1`] = `
<div>
  <h1
    className="Logo_title_229208"
  >
    <a
      className="Logo_link_18deiwy u-flex"
      href="/"
      onClick={[Function]}
    >
      <svg />
      GitHub Repo Search
    </a>
  </h1>
  <p
    className="Logo_subTitle_1r70osc"
  >
    Browse repos via 
    <a
      href="https://developer.github.com/v3/"
    >
      the GitHub API
    </a>
  </p>
</div>
`;
